packages/compiler/server/serverlib.ts

```diff
@@ -0,0 +1,698 @@
+      diagnostics.push(diagnostic);
+    }
+
+    for (const [document, diagnostics] of diagnosticMap) {
+      sendDiagnostics(document, diagnostics);
+    }
+  }
+
+  async function gotoDefinition(params: DefinitionParams): Promise<Location | undefined> {
+    const document = host.getDocumentByURL(params.textDocument.uri);
+    if (!document) {
+      return undefined;
+    }
+
+    const program = await compile(document);
+    if (!program) {
+      return undefined;
+    }
+
+    const path = getPath(document);
+    const file = program.sourceFiles.get(path);
+    if (!file) {
+      return undefined;
+    }
+
+    const referringNode = getNodeAtPosition(
+      file,
+      document.offsetAt(params.position),
+      (node) => node.kind === SyntaxKind.TypeReference
+    );
+    if (!referringNode) {
+      return undefined;
+    }
+
+    const definingNode = program.checker?.getTypeForNode(referringNode)?.node;
+    if (!definingNode) {
+      return undefined;
+    }
+
+    const definingLocation = getSourceLocation(definingNode);
+    return convertSourceLocation(definingLocation);
+  }
+
+  async function complete(params: CompletionParams): Promise<CompletionList> {
+    const completions: CompletionList = {
+      isIncomplete: false,
+      items: [],
+    };
+
+    const document = host.getDocumentByURL(params.textDocument.uri);
+    if (!document) {
+      return completions;
+    }
+
+    const program = await compile(document);
+    if (!program) {
+      return completions;
+    }
+
+    const path = getPath(document);
+    const file = program.sourceFiles.get(path);
+    if (!file) {
+      return completions;
+    }
+
+    const node = getNodeAtPosition(file, document.offsetAt(params.position));
+    if (node === undefined) {
+      addKeywordCompletion("root", completions);
+    } else {
+      switch (node.kind) {
+        case SyntaxKind.NamespaceStatement:
+          addKeywordCompletion("namespace", completions);
+          break;
+        case SyntaxKind.Identifier:
+          addIdentifierCompletion(program, node, completions);
+          break;
+      }
+    }
+
+    return completions;
+  }
+
+  function addKeywordCompletion(area: keyof KeywordArea, completions: CompletionList) {
+    const filteredKeywords = keywords.filter(([_, x]) => area in x);
+    for (const [keyword] of filteredKeywords) {
+      completions.items.push({
+        label: keyword,
+        kind: CompletionItemKind.Keyword,
+      });
+    }
+  }
+
+  /**
+   * Add completion options for an identifier.
+   */
+  function addIdentifierCompletion(
+    program: Program,
+    node: IdentifierNode,
+    completions: CompletionList
+  ) {
+    const result = program.checker!.resolveCompletions(node);
+    if (result.size === 0) {
+      return;
+    }
+    for (let [key, { sym, label }] of result) {
+      let documentation: string | undefined;
+      let kind: CompletionItemKind;
+      if (sym.kind === "using") {
+        sym = sym.symbolSource;
```

Comment:
No, it fixes an issue where we would get function icon for anything pulled in via using. Previously symbol was either type or decorator, but now it can be using and we need to get the symbol source to pick the right icon.